docs/user-guide/kubectl-overview.md

```diff
@@ -33,160 +33,271 @@ Documentation for other releases can be found at
+TODO: Auto-generate this file to ensure it's always in sync with any `kubectl` changes, see [#14177](https://github.com/kubernetes/kubernetes/pull/14177).
 
-## Overview
+## Syntax
 
-`kubectl` controls the Kubernetes cluster manager. The synopsis is:
+Use the following syntax to run `kubectl` commands from your terminal window:
 
 ```
 kubectl [command] [TYPE] [NAME] [flags]
 ```
 
-This specifies:
+where `command`, `TYPE`, `NAME`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources, for example `create`, `get`, `describe`, `delete`.
+* `TYPE`: Specifies the [resource type](#resource-types). Resource types are case-sensitive and both the singular and plural forms are valid. For example, the following two commands produce the same output:
+
+   ```
+    $ kubectl get pod pod1
+    $ kubectl get pods pod1
+   ```
+
+* `NAME`: Specifies the name of the resource. Names are case-sensitive. If the name is omitted, details for all resources are displayed, for example `$ kubectl get pods`.
+
+   When performing an operation on multiple resources, you can specify each resource by type and name or specify one or more files:
+   * To specify resources by type and name:
+	    * To group resources if they are all the same type: `TYPE1 name1 name2 name<#>`<br/>
+        Example: `$ kubectl get pod example-pod1 example-pod2`
+        * To specify multiple resource types individually: `TYPE1/name1 TYPE1/name2 TYPE2/name3 TYPE<#>/name<#>`<br/>
+        Example: `$ kubectl get pod/example-pod1 replicationcontroller/example-rc1`
+   * To specify resources with one or more files: `-f file1 -f file2 -f file<#>`
+	 [Use YAML rather than JSON](config-best-practices.md) since YAML tends to be more user-friendly, especially for configuration files.<br/>
+     Example: `$ kubectl get pod -f ./pod.yaml`
+* `flags`: Specifies optional flags. For example, you can use the `-s` or `--server` flags to specify the address and port of the Kubernetes API server.<br/>
+**Important**: Flags that you specify from the command line override default values and any corresponding environment variables. *Tip*: Use short flags sparingly and only for the most frequently used options.<br/>
+For more information and examples about specifying flags, see the [kubectl conventions](../devel/kubectl-conventions.md#flag-conventions).
+
+If you need help, just run `kubectl help` from the terminal window.
+
+## Operations
+
+The following table includes short descriptions and the general syntax for all of the `kubectl` operations:
+
+Operation       | Syntax	|       Description
+-------------------- | -------------------- | --------------------
+`annotate`	| `kubectl annotate [--overwrite] (-f FILENAME | TYPE NAME) KEY_1=VAL_1 ... KEY_N=VAL_N [--resource-version=version]` | Add or update the annotations of a resource.
+`api-versions`	| `kubectl api-versions` | List the API versions in use.
```

Comment:
Changed to:
"List the API versions that are available."

I initially changed this because I thought its not whats "available" but rather whats been "installed"/ in use. 

Also added: `[flags]`